omod/src/test/java/org/openmrs/module/fhir2/narrative/MedicationRequestNarrativeTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.narrative;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.narrative.CustomThymeleafNarrativeGenerator;
+import ca.uhn.fhir.parser.IParser;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.junit.Test;
+import org.openmrs.module.fhir2.FhirConstants;
+
+public class MedicationRequestNarrativeTest {
+	
+	private static final String EXAMPLE_RESOURCE_PATH = "org/openmrs/module/fhir2/narrative/example/medicationrequest-example.json";
+	
+	private static final String EXPECTED_NARRATIVE_PATH = "org/openmrs/module/fhir2/narrative/expected/medicationrequest-expected.html";
+	
+	@Test
+	public void shouldGenerateAMedicationRequestNarrative() throws IOException {
+		FhirContext ctx = FhirContext.forR4();
+		ctx.setNarrativeGenerator(new CustomThymeleafNarrativeGenerator(FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE,
+		        FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE));
+		IParser parser = ctx.newJsonParser();
+		
+		MedicationRequest given = parser.parseResource(MedicationRequest.class,
+		    getClass().getClassLoader().getResourceAsStream(EXAMPLE_RESOURCE_PATH));
+		MedicationRequest result = parser.parseResource(MedicationRequest.class, parser.encodeResourceToString(given));
+		String expected = IOUtils.readLines(getClass().getClassLoader().getResourceAsStream(EXPECTED_NARRATIVE_PATH),
+		    Charsets.toCharset(Charset.defaultCharset())).get(0);
```

Comment:
@ibacher PTAL. Doing it this way is fine right? 
Using just `equalTo(getClass().getClassLoader().getResourceAsStream(<path>)` raised errors since it returns a stream.

Reply:
Why not just use `IOUtils.resourceToString(EXPECTED_NARRATIVE_PATH, StandardCharsets.UTF_8)`? 

Also, some general points about unit tests. We want to structure unit tests so that it is clear what is being tested in each test. We also want to factor out the common setup parts so that things aren't repeating themselves too much.

For unit tests, I use this structure:

```java
// given - setup the preconditions

// when - do the thing being tested

// then - assertions to ensure state is right
```

Given this structure, I think this test should look something like this:

```java
// do this once per class
private FhirContext ctx = FhirContext.forR4();

private IParser parser;

@Before
private void setup() {
    ctx.setNarrativeGenerator(new CustomThymeleafNarrativeGenerator(FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE,
        FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE));
    parser = ctx.newJsonParser();
}

@Test
public void shouldGenerateMedicationRequestNarrative() throws IOException {
    // given
    MedicationRequest given = parser.parseResource(MedicationRequest.class,
         getClass().getClassLoader().getResourceAsStream(EXAMPLE_RESOURCE_PATH));

    // when
    MedicationRequest result = parser.parseResource(MedicationRequest.class, parser.encodeResourceToString(given));

    // then
    assertThat(result, notNullValue());
    assertThat(result.getText(), notNullValue());
    assertThat(result.getText().getStatusAsString(), equalTo("generated"));
    // if you don't do it in one line like this, I would load the expected narrative here
    // that way, if some error comes from just loading the expected result, we can at least be
    // assured that the other assertions passed
    assertThat(result.getText().getDivAsString(),
        equalTo(IOUtils.resourceToString(EXPECTED_NARRATIVE_PATH, StandardCharsets.UTF_8)));
}
```

The comments there don't need to be preserved. They're just to show you why I've put the line breaks where I did.

Reply:
Okay! Will restructure the tests this way.

Reply:
Actually I used `IOUtils.readlines()` because I just needed the first line, since I had to keep a newline in the end. Though, I guess I can use `IOUtils.resourceToString()` too and then split it at a newline.

Okay, will restructure the tests.

Reply:
@ibacher I've done the changes. PTAL.

Reply:
Done the changes.